### PR TITLE
Add Limine bootloader (closes #973)

### LINKED ISSUE #973: Add Limine bootloader
Rational:

> Tool kits available on both Windows and Linux
> Supports 64bit long model
> Supports GTP
> Supports multiple file systems

More info:

https://github.com/limine-bootloader/limine

### PR DESCRIPTION
**Pull request by AnErrupTion**
This PR adds the Limine bootloader!
Note: The ``limine-deploy`` tool has been ported to a single C# class to avoid using external tools!

### CODE REVIEW

File: Source/Mosa.Utility.BootImage/Limine.cs [hunk 1]
```diff
@@ -0,0 +1,1618 @@
+﻿using System;
+using System.Linq;
+using System.IO;
+
+namespace Mosa.Utility.BootImage
+{
+	//https://github.com/limine-bootloader/limine/blob/v3.0-binary/limine-deploy.c
+	public static class Limine
+	{
+		private static byte[] bootloader_img = {
```
Comment (tgiphil) [inline Source/Mosa.Utility.BootImage/Limine.cs:10]: Include the binary as resource instead.

File: Source/Mosa.Utility.BootImage/Limine.cs [hunk 2]
```diff
@@ -0,0 +1,1618 @@
...
+			Write(bootloader_img, 0, 512);
+
+			// Write the rest of stage 2 to the device
+			Write(bootloader_img.Skip(512).ToArray(), stage2LocA, stage2SizeA);
+			Write(bootloader_img.Skip(512 + stage2SizeA).ToArray(), stage2LocB, (ulong)(stage2Size - stage2SizeA));
+
+			// Hardcode in the bootsector the location of stage 2 halves
+			// It seems like we need to use pointers here
+			stage2SizeA = EndSwap(stage2SizeA);
+			WritePtr(&stage2SizeA, 0x1a4, sizeof(ushort));
```
Comment (tgiphil) [inline Source/Mosa.Utility.BootImage/Limine.cs:1455]: No unsafe code! 

See Mosa.Utility.BootImage\Syslinux.cs and Generator.cs as examples. 

Note: We are not supporting big endian (all the code to do that was removed a few years ago).

File: Source/Mosa.Utility.Launcher/Builder.cs [hunk 3]
```diff
@@ -199,30 +200,38 @@ private void CreateDiskImage(string imagefile)
 		{
 			var bootImageOptions = new BootImageOptions();
 
-			if (LauncherSettings.ImageBootLoader == "syslinux3.72")
+			if (LauncherSettings.ImageBootLoader.StartsWith("syslinux"))
```
Comment (tgiphil) [inline Source/Mosa.Utility.Launcher/Builder.cs:203]: Let's be exact and use a switch/case.

File: Tools/limine/limine.cfg [hunk 4]
```diff
@@ -0,0 +1,6 @@
+TIMEOUT=5
```
Comment (tgiphil) [inline Tools/limine/limine.cfg:1]: Future. Let's create the config file within Starter.cs so it can be customized later.

**Review by tgiphil — CHANGES_REQUESTED** [review]

File: Source/Mosa.Utility.BootImage/Limine.cs [hunk 5]
```diff
@@ -0,0 +1,1618 @@
...
+
+				for (ulong i = 0; i < chunk; i++)
+					Cache[offset + i] = buffer[progress + i];
+
+				State = CacheState.Dirty;
+				progress += chunk;
+			}
+		}
+
+		private static unsafe void WritePtr(void* _buffer, ulong loc, ulong count)
```
Comment (tgiphil) [inline Source/Mosa.Utility.BootImage/Limine.cs]: Unsafe code!

File: Source/Mosa.Utility.BootImage/Limine.cs (hunk 1 above)
Comment (tgiphil) [inline Source/Mosa.Utility.BootImage/Limine.cs]: Include the binary as resource instead.

**Review by tgiphil — CHANGES_REQUESTED** [review]

File: Source/Mosa.Utility.BootImage/Limine.cs (hunk 1 above)
Comment (tgiphil) [inline Source/Mosa.Utility.BootImage/Limine.cs]: Include the binary as resource instead.

File: Source/Mosa.Utility.BootImage/Limine.cs [hunk 6]
```diff
@@ -0,0 +1,1618 @@
...
+		enum CacheState { Clean, Dirty }
+
+		private static FileStream Stream;
+		private static byte[] Cache;
+		private static ulong CachedBlock;
+		private static CacheState State;
+
+		private const int BlockSize = 512;
+
+		public static unsafe void Deploy(string path)
```
Comment (tgiphil) [inline Source/Mosa.Utility.BootImage/Limine.cs]: Unsafe code!

File: Source/Mosa.Utility.BootImage/Limine.cs (hunk 5 above)
Comment (tgiphil) [inline Source/Mosa.Utility.BootImage/Limine.cs]: Unsafe code!

**Review by tgiphil — CHANGES_REQUESTED** [review]

**Review by charsleysa — APPROVED** [review]
LGTM

**Pull request merged**